internal/provider/resource_gitlab_project_runner.go

```diff
@@ -0,0 +1,120 @@
+package provider
+
+import (
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+var _ = registerResource("gitlab_project_runner", func() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGitlabProjectRunnerCreate,
+		Read:   resourceGitlabProjectRunnerRead,
+		Delete: resourceGitlabProjectRunnerDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"project": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"runner_id": {
+				Type:     schema.TypeInt,
+				ForceNew: true,
+				Required: true,
+			},
+		},
+	}
+})
+
+func resourceGitlabProjectRunnerCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gitlab.Client)
+	projectID := d.Get("project").(string)
+	runnerID := d.Get("runner_id").(int)
+	options := &gitlab.EnableProjectRunnerOptions{
+		RunnerID: runnerID,
+	}
+
+	log.Printf("[DEBUG] create gitlab project runner %s/%v", projectID, runnerID)
+
+	_, resp, err := client.Runners.EnableProjectRunner(projectID, options)
+	// TODO: Implement proper error handling with parsin of response message
+	// Ignore error if response contains "{message: {assign_to: [Validation failed: Runner has already been taken]}}"
+	if err != nil && resp.StatusCode != 400 {
+		return err
+	}
+
+	runnerIDString := strconv.Itoa(runnerID)
+	d.SetId(buildTwoPartID(&projectID, &runnerIDString))
+
+	return resourceGitlabProjectRunnerRead(d, meta)
+}
+
+func resourceGitlabProjectRunnerRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gitlab.Client)
+	runnerType := "project_type"
+	projectID, runnerID, err := projectAndRunnerFromID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	options := &gitlab.ListProjectRunnersOptions{
+		Type: &runnerType,
```

Comment:
We'll want to look at this value, since it's filtering by "project_type" here, and that's a deprecated feature in GitLab: https://docs.gitlab.com/ee/update/deprecations.html#rest-api-endpoint-to-list-group-runners-no-longer-accepts-project_type-value-for-type-argument

Is there a better way we should be doing this?

Reply:
Looks like this deprecation is only for the `/groups/` endpoint. We're using it here on the `/projects/` endpoint where it still makes sense and as far as I can tell it is not deprecated here.

Reply:
Good catch!